src/nvim/api/private/helpers.h

```diff
@@ -8,15 +8,10 @@
 #include "nvim/memory.h"
 #include "nvim/lib/kvec.h"
 
+void _api_set_error(Error *err, ErrorType errType, const char *format, ...);
 // -V:api_set_error:618
 #define api_set_error(err, errtype, ...) \
-  do { \
-    snprintf((err)->msg, \
-             sizeof((err)->msg), \
-             __VA_ARGS__); \
-    (err)->set = true; \
-    (err)->type = kErrorType##errtype; \
-  } while (0)
+  _api_set_error(err, kErrorType##errtype, __VA_ARGS__)
```

Comment:
The macro should be all-caps. So the function could be used without leading underscore. And there is no need to duplicate declaration, macro is not used before including `api/private/helpers.h.generated.h`.